pkgs/tools/backup/restic/default.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, lib, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
+  name = "restic-${version}";
+  version = "0.5.0";
+
+  goPackagePath = "github.com/restic/restic";
+
+  src = fetchFromGitHub {
+    owner = "restic";
+    repo = "restic";
+    rev = "v${version}";
+    sha256 = "0dj6zg4b00pwgs6nj7w5s0jxm6cfavd9kdcq0z4spypwdf211cgl";
+  };
+
+  buildPhase = ''
```

Comment:
Without the custom `buildPhase`, lots of errors like this one occur:

```
go/src/github.com/restic/restic/src/restic/repository/packer_manager.go:17:2: cannot find package "restic/pack" in any of:
	/tmp/nix-build-restic-0.5.0.drv-0/go/src/github.com/restic/restic/vendor/restic/pack (vendor tree)
	/nix/store/j8hhdgfxgavib7ng7d7kkqj2kpjrxprh-go-1.7.4/share/go/src/restic/pack (from $GOROOT)
	/tmp/nix-build-restic-0.5.0.drv-0/go/src/restic/pack (from $GOPATH)
```

Also, the project's build instructions say building should be done like I did in the custom `buildPhase`.

Without the custom `installPhase`, I'm getting

```
output ‘/nix/store/dc5i8lz5yfw6z7mp0187farpgmnj6vji-restic-0.5.0’ is not allowed to refer to path ‘/nix/store/j8hhdgfxgavib7ng7d7kkqj2kpjrxprh-go-1.7.4’
```